### PR TITLE
Feature/separate robot skills

### PR DESCRIPTION
**Pull request by jlunenburg**

### CODE REVIEW

File: amigo_skills/package.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <depend>amigo_description</depend>
+  <depend>robot_skills</depend>
+
+  <!-- The export tag contains other, unspecified, tags -->
+  <export>
+    <!-- Other tools can request additional information be placed here -->
+
+  </export>
```
Comment (MatthijsBurgh) [inline amigo_skills/package.xml:20]:
```suggestion
  <buildtool_depend>catkin</buildtool_depend>

  <depend>amigo_description</depend>
  <depend>robot_skills</depend>

```

File: amigo_skills/test/test_amigo.test [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" ?>
```
Comment (MatthijsBurgh) [inline amigo_skills/test/test_amigo.test:1]:
```suggestion
<?xml version="1.0"?>
```

File: amigo_skills/test/test_amigo.test [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+<launch>
+    <!-- N.B.: test-name MUST be amigo in order for namespacing to work -->
```
Comment (MatthijsBurgh) [inline amigo_skills/test/test_amigo.test:3]: This kind of assumptions should be documented in `robot_skills`

File: challenge_final/src/challenge_final/navigate_to_and_grab_rack.py [hunk 4]
```diff
@@ -132,6 +132,6 @@ def __init__(self, robot, rack_id, rack_navigation_area):
 
 if __name__ == '__main__':
     rospy.init_node(os.path.splitext("test_" + os.path.basename(__file__))[0])
-    hero = Hero()
-    hero.reset()
-    NavigateToAndGrabRack(hero, "rack", "in_front_of").execute()
+    robot_instance = get_robot("hero")
```
Comment (MatthijsBurgh) [inline challenge_final/src/challenge_final/navigate_to_and_grab_rack.py:135]: I don't like this. Do you really think that using `robot` would be too unclear/confusing?

File: challenge_set_the_table/src/challenge_set_the_table/navigate_to_and_place_item_on_table.py [hunk 5]
```diff
@@ -253,23 +248,25 @@ def _publish_item_poses(robot, items):
...
 
-    hero = Hero()
-    _publish_item_poses(hero, item_poses)
-    hero.reset()
+    robot_instance = get_robot("hero")
+    _publish_item_poses(robot_instance, item_poses)
+    robot_instance.reset()
     try:
-        placement_height = float(sys.argv[2])
+        placement_height_ = float(sys.argv[2])
```
Comment (MatthijsBurgh) [inline challenge_set_the_table/src/challenge_set_the_table/navigate_to_and_place_item_on_table.py:264]: Any reason for the trailing underscore? I know in C++ is is common to do with private class variables, but that isn't the case here.

File: hero_skills/package.xml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  <depend>hero_description</depend>
+  <depend>robot_skills</depend>
+
+  <test_depend>rostest</test_depend>
+
+  <!-- The export tag contains other, unspecified, tags -->
+  <export>
+    <!-- Other tools can request additional information be placed here -->
+
+  </export>
```
Comment (MatthijsBurgh) [inline hero_skills/package.xml:23]:
```suggestion
```

File: hero_skills/src/hero_skills/hero.py [hunk 7]
```diff
@@ -24,9 +27,13 @@ def __init__(self, wait_services=False):
 
         # add hero's arm
         hero_arm = arms.Arm(self.robot_name, self.tf_listener, self.get_joint_states, "left")
-        hero_arm.add_part('force_sensor', force_sensor.ForceSensor(self.robot_name, self.tf_listener, "/" + self.robot_name + "/wrist_wrench/raw"))
+        hero_arm.add_part('force_sensor', force_sensor.ForceSensor(
+            self.robot_name, self.tf_listener, "/" + self.robot_name + "/wrist_wrench/raw"
```
Comment (MatthijsBurgh) [inline hero_skills/src/hero_skills/hero.py:31]:
```suggestion
            self.robot_name, self.tf_listener, self.robot_name + "/wrist_wrench/raw"
```
Frames should not start with leading slash

File: robot_skills/src/robot_skills/get_robot.py [hunk 8]
```diff
@@ -4,19 +4,29 @@
...
 from .robot import Robot
-from .sergio import Sergio
 
-ROBOTS = {
-    "amigo": Amigo,
-    "hero": Hero,
-    "mockbot": Mockbot,
-    "sergio": Sergio
-}
+# ToDo: create a decent registration here
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/get_robot.py:10]: Create an issue for this

File: robot_skills/src/robot_skills/test_tools/test_robot.py [hunk 9]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (MatthijsBurgh) [inline robot_skills/src/robot_skills/test_tools/test_robot.py:1]: Missing shebang?

File: sergio_skills/package.xml [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <depend>sergio_description</depend>
+  <depend>robot_skills</depend>
+
+  <!-- The export tag contains other, unspecified, tags -->
+  <export>
+    <!-- Other tools can request additional information be placed here -->
+
+  </export>
```
Comment (MatthijsBurgh) [inline sergio_skills/package.xml:20]:
```suggestion
  <buildtool_depend>catkin</buildtool_depend>

  <depend>robot_skills</depend>
  <depend>sergio_description</depend>

```

File: sergio_skills/package.xml [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (MatthijsBurgh) [inline sergio_skills/package.xml:2]:
```suggestion
<?xml-model
  href="http://download.ros.org/schema/package_format3.xsd"
  schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
```

File: hero_skills/package.xml [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (MatthijsBurgh) [inline hero_skills/package.xml:2]:
```suggestion
<?xml-model
  href="http://download.ros.org/schema/package_format3.xsd"
  schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
```

File: amigo_skills/package.xml [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<package format="2">
```
Comment (MatthijsBurgh) [inline amigo_skills/package.xml:2]:
```suggestion
<?xml-model
  href="http://download.ros.org/schema/package_format3.xsd"
  schematypens="http://www.w3.org/2001/XMLSchema"?>
<package format="3">
```

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
See comments

Plus right now the XXX_description packages are added as `<depend>`. Though these are only used in the test. So I think these should be changed to a test dependency. In that case also the CMakeLists should be updated.

File: amigo_skills/test/test_amigo.test (hunk 3 above)
Comment (jlunenburg) [reply amigo_skills/test/test_amigo.test:3]: Good point, no need for people to reinvent the wheel. See f96f2719fbde0d56b7e0774b22becbe009a44bd3 (was the best location I could think of)

File: challenge_final/src/challenge_final/navigate_to_and_grab_rack.py (hunk 4 above)
Comment (jlunenburg) [reply challenge_final/src/challenge_final/navigate_to_and_grab_rack.py:135]: The problem is that this is in the same file. As a result, you get a warning that names shadow a variable in the outer scope (which I consider bad practice). Therefore, I want to use a different name and this seems suitable to me.

File: challenge_set_the_table/src/challenge_set_the_table/navigate_to_and_place_item_on_table.py (hunk 5 above)
Comment (jlunenburg) [reply challenge_set_the_table/src/challenge_set_the_table/navigate_to_and_place_item_on_table.py:264]: Same reason as before, only to differentiate the variable name from the 'placement_height' that is used in the state.

File: hero_skills/package.xml (hunk 6 above)
Comment (jlunenburg) [reply hero_skills/package.xml]: See 0c711187e70e90b6f544dc703ebd8cc5424a9d1a

File: hero_skills/src/hero_skills/hero.py (hunk 7 above)
Comment (jlunenburg) [reply hero_skills/src/hero_skills/hero.py]: Sharp, didn't introduce it but shouldn't be there. Fix in 7c24ed8e2f28d950b0b298e107bdd896c011e9b9

File: robot_skills/src/robot_skills/test_tools/test_robot.py (hunk 9 above)
Comment (jlunenburg) [reply robot_skills/src/robot_skills/test_tools/test_robot.py:4]: Why? The purpose of this is that deriving packages can use this class in their tests

File: robot_skills/src/robot_skills/get_robot.py (hunk 8 above)
Comment (jlunenburg) [reply robot_skills/src/robot_skills/get_robot.py:10]: See https://github.com/tue-robotics/tue_robocup/issues/1095

File: sergio_skills/package.xml (hunk 10 above)
Comment (jlunenburg) [reply sergio_skills/package.xml]: See f96f2719fbde0d56b7e0774b22becbe009a44bd3

File: sergio_skills/package.xml (hunk 11 above)
Comment (jlunenburg) [reply sergio_skills/package.xml]: See f96f2719fbde0d56b7e0774b22becbe009a44bd3

File: hero_skills/package.xml (hunk 12 above)
Comment (jlunenburg) [reply hero_skills/package.xml]: See f96f2719fbde0d56b7e0774b22becbe009a44bd3

File: amigo_skills/package.xml (hunk 13 above)
Comment (jlunenburg) [reply amigo_skills/package.xml]: See f96f2719fbde0d56b7e0774b22becbe009a44bd3

File: robot_skills/src/robot_skills/test_tools/test_robot.py (hunk 9 above)
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/test_tools/test_robot.py:4]: I thought this should be executable. But a second look proves me wrong.

**Comment by LoyVanBeek** [comment]
Looks good guys! 

Is this introducing a cyclic dependency now btw? From eg. hero_skills to robot_skills for the robot parts and from robot_skills to hero_skills for the `get_robot` function?

File: amigo_skills/package.xml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  <maintainer email="PetervDooren@outlook.com">Peter van Dooren</maintainer>
+
+  <license>BSD</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <depend>robot_skills</depend>
+
+  <test_depend>rostest</test_depend>
+  <test_depend>amigo_description</test_depend>
```
Comment (MatthijsBurgh) [inline amigo_skills/package.xml:19]:
```suggestion
  <test_depend>amigo_description</test_depend>
  <test_depend>rostest</test_depend>
```

File: hero_skills/package.xml [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+  <maintainer email="PetervDooren@outlook.com">Peter van Dooren</maintainer>
+
+  <license>BSD</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <depend>robot_skills</depend>
+
+  <test_depend>rostest</test_depend>
+  <test_depend>hero_description</test_depend>
```
Comment (MatthijsBurgh) [inline hero_skills/package.xml:19]:
```suggestion
  <test_depend>hero_description</test_depend>
  <test_depend>rostest</test_depend>
```

File: sergio_skills/package.xml [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+  <maintainer email="PetervDooren@outlook.com">Peter van Dooren</maintainer>
+
+  <license>BSD</license>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <depend>robot_skills</depend>
+
+  <test_depend>rostest</test_depend>
+  <test_depend>sergio_description</test_depend>
```
Comment (MatthijsBurgh) [inline sergio_skills/package.xml:19]:
```suggestion
  <test_depend>sergio_description</test_depend>
  <test_depend>rostest</test_depend>
```

**Review by MatthijsBurgh — APPROVED** [review]

**Pull request merged**